### PR TITLE
Add Combobox component

### PR DESCRIPTION
**Pull request by balinterdi**
Co-authored-by: David McNamara <david.mcnamara@phorest.com>

The original implementation by done by David but the file structure of the add-on has been drastically changed so the feasible way was to copy his work from the previous place.

Also, Ember seems to have gotten way stricter about throwing the "You attempted to update activeOptionIndex but it'd already been used previously in the same computation" errors so a lot of the pieces needed to be refactored to make tests pass.

### CODE REVIEW

File: ember-headlessui/addon/components/combobox.hbs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    Options=(component
+      'combobox/-options'
+      isOpen=this.isOpen
+      guid=this.guid
+      registerOptionElement=this.registerOptionElement
+      unregisterOptionElement=this.unregisterOptionElement
+      registerOptionsElement=this.registerOptionsElement
+      hasLabelElement=this.labelElement
+      activeOptionGuid=this.activeOptionGuid
+      activeOptionIndex=this.activeOptionIndex
```
Comment (dmcnamara-eng) [inline ember-headlessui/addon/components/combobox.hbs:16]: ✂️ This is unused now?

File: ember-headlessui/addon/components/combobox/-options.hbs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      ...attributes
+    >
+      {{yield
+        (hash
+          Option=(component
+            'combobox/-option'
+            registerOptionElement=@registerOptionElement
+            unregisterOptionElement=@unregisterOptionElement
+            activeOptionGuid=@activeOptionGuid
+            activeOptionIndex=@activeOptionIndex
```
Comment (dmcnamara-eng) [inline ember-headlessui/addon/components/combobox/-options.hbs:34]: ✂️ This is unused now?

**Comment by dmcnamara-eng** [comment]
Somethings not quite right here.

I should be able to hit enter to select the first item in the list.

Compare this implementation compared to [react](http://localhost:4200/combobox/combobox-basic).

https://user-images.githubusercontent.com/70960245/193063249-1caeb917-cb50-4166-aea5-f3428662218e.mov

https://user-images.githubusercontent.com/70960245/193063524-43a1f502-c543-464f-bd09-d3e0932292ad.mov

**Comment by balinterdi** [comment]
> I should be able to hit enter to select the first item in the list.

Indeed, that's not covered by tests, I'll take a look.

**Comment by balinterdi** [comment]
@rreckonerr The challenge with using a branch (custom SHA) of a forked *monorepo* is that the top-level folder of the github repository is not the same as the package's folder. Here, the package lives under `/ember-headlessui` in the repo.

The solution I've found is to use a service called gitpkg that allows to "link" to a subfolder in the dependency string.

Here is what we use as the dependency for `ember-headlessui` in our package.json:

> https://gitpkg.now.sh/phorest/ember-headlessui/ember-headlessui?028782f7e65d8942c4d92d0e5df42f97c3b96fc0

That SHA at the end might change but you get the point. (You can even create a fork in your repo to have more control over the version you use).

**Comment by rreckonerr** [comment]
@balinterdi Thanks for the quick response! Do you also use pnpm for that? I think I hit [this issue](https://github.com/pnpm/pnpm/issues/3886) when I tried to install it from the gitpkg the last time

**Comment by NullVoxPopuli** [comment]
This project uses pnpm

**Comment by balinterdi** [comment]
@NullVoxPopuli Thank you for explaining.

I meant that the host project that uses this fork of ember-headlessui uses yarn.

I first tried to yarn link it locally to test but it didn't seem to work at all – the ember-headlessui component files didn't make it into the build of the host app. After trying in vain for half a day, I found gitpkg and it worked on the first try.

**Comment by dmcnamara-eng** [comment]
Thanks for updating this @balinterdi 

Seems to be API complete at this point and robust enough to merge. 👍 ⭐ 💥

**Pull request merged**